packages/@sanity/schema/src/sanity/extractSchema.ts

```diff
@@ -0,0 +1,669 @@
+    key: 'children',
+    value: optionalField({
+      type: 'array',
+      of: {
+        type: 'union',
+        of: [
+          {
+            type: 'object',
+            fields: [
+              createKeyField(),
+              {
+                type: 'objectKeyValue',
+                key: 'text',
+                value: optionalField({
+                  type: 'string',
+                }),
+              } satisfies ObjectKeyValueField<OptionalField<StringField>>,
+              {
+                type: 'objectKeyValue',
+                key: 'marks',
+                value: optionalField({
+                  type: 'array',
+                  of: {
+                    type: 'union',
+                    of: marks,
+                  },
+                }),
+              } satisfies ObjectKeyValueField<OptionalField<ArrayField<UnionField>>>,
+            ],
+          } satisfies ObjectField,
+        ],
+      } satisfies UnionField<ObjectField>,
+    }),
+  } satisfies ObjectKeyValueField<OptionalField<ArrayField<UnionField<ObjectField>>>>
+
+  const markDefsField: ObjectKeyValueField<
+    OptionalField<ArrayField<UnionField<ObjectField | ReferenceField | UnionField<ReferenceField>>>>
+  > = {
+    type: 'objectKeyValue',
+    key: 'markDefs',
+    value: optionalField({
+      type: 'array',
+      of: {
+        type: 'union',
+        of:
+          blockDefinition.marks?.annotations?.map((annotation) => createMarkDefField(annotation)) ||
+          [],
+      },
+    }),
+  }
+  return {
+    type: 'object',
+    fields: [levelField, styleField, listItemField, childrenField, markDefsField, createKeyField()],
+  }
+}
+
+function createMarkDefField(
+  annotation: ArrayOfType<'object' | 'reference'>,
+): ObjectField | ReferenceField | UnionField<ReferenceField> {
+  if (annotation.type === 'object' && 'fields' in annotation) {
+    const fields: ObjectKeyValueField[] = [
+      createKeyField(),
+      ...annotation.fields.map(
+        (field): ObjectKeyValueField => ({
+          type: 'objectKeyValue',
+          key: field.name,
+          value: optionalField(parseField(field)),
+        }),
+      ),
+    ]
+
+    return {
+      type: 'object',
+      fields,
+    }
+  }
+
+  if (annotation.type === 'reference' && 'to' in annotation) {
+    return createReferenceFieldDefintion(annotation)
+  }
+
+  return {
+    type: 'object',
+    fields: [],
+  }
+}
+
+function createReferenceFieldDefintion(
+  reference: Pick<ReferenceDefinition, 'to'>,
+): ReferenceField | UnionField<ReferenceField> {
+  if (Array.isArray(reference.to)) {
+    return {
+      type: 'union',
+      of: reference.to.map((t) => ({
+        type: 'reference',
+        to: t.type,
+      })),
+    }
+  }
+  return {
+    type: 'reference',
+    to: reference.to.type,
+  } satisfies ReferenceField
+}
+function createCrossDatasetReferenceFieldDefintion(_: CrossDatasetReferenceDefinition): Field {
+  return {type: 'unknown'} satisfies UnknownField
```

Comment:
why do we treat it as unknow? out of scope this time around?

Reply:
Yep, we don't support it, yet. Would increase the scope to need to handle multiple schemas when parsing queries.